Request OPTIONS fix to return all allowed methods

michalbundyra opened the pull request:
The integration tests are pulled from zend-expressive-router repository.

See: https://github.com/zendframework/zend-expressive-router/pull/53

michalbundyra committed:
Fixed broken tests

The interface of the latest zend-expressive-router has been changed.
We can't get matched middleware from RouteResult but we can get matched
route, and then from route we can get middleware.

michalbundyra committed:
Added ImplicitOptionsMiddlewareIntegrationTest

The test is pulled from zend-expressive-router library
OPTIONS request should return response with all allowed methods for
the matching path

michalbundyra committed:
AuraRouter returns route with all allowed method on OPTIONS request

michalbundyra committed:
OPTIONS request returns failure and RouteResult contains allowed methods

michalbundyra committed:
Updated integration tests - HEAD and OPTIONS requests

michalbundyra committed:
Removed unused imports

michalbundyra committed:
Use constant instead of string

michalbundyra committed:
Updated integration tests - explicit HEAD and OPTIONS requests

geerteltink commented on composer.json:
```diff
@@ -20,17 +20,24 @@
+    "repositories": [
+        {
+            "type": "git",
+            "url": "https://github.com/webimpress/zend-expressive-router"
```
A reminder to remove this before merging.

michalbundyra committed:
HEAD request can be done done on route with GET method allowed

weierophinney committed:
Updates to zend-expressive-router 3.0.0rc1

Pins to 3.0.0rc1, and updates new test case based on name change in that
release.
